### PR TITLE
Run unit tests in debug mode

### PR DESCRIPTION
**Pull request by branan**
# Description

Removes `--release` from our testing script for running unit tests. On my laptop this is 8 minutes faster. In CI where there is less parallelism, I expect that number to be larger.

## Type of change

- [x] CI cleanup

# How Has This Been Tested?

Ran tests locally

# Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I rebased on the latest `main` branch

### CODE REVIEW

**Comment by lemunozm** [comment]
Consider merging https://github.com/centrifuge/centrifuge-chain/pull/1239 into this PR before

File: docs/DEVELOPING.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ and another one to verify how it works in a more real environment as a parachain
 The following command will run the unit and integration tests:
 
 ```bash
-cargo +nightly test --workspace --release --features runtime-benchmarks,try-runtime
```
Comment (mustermeiszer) [inline docs/DEVELOPING.md:31]: Why is this removed? Is it only required for the node build?

Comment (lemunozm) [reply docs/DEVELOPING.md:31]: `--release` only for production nodes or benchmarks.
`+nightly` is set automatically by `rust-toolchain.toml`

**Review by mustermeiszer — DISMISSED** [review]

**Comment by mikiquantum** [comment]
@lemunozm try adding the following to the test.yml: 
https://github.com/centrifuge/centrifuge-chain/blob/009256a0a8b3be00d9c3b3cc6e794abf54da7a6d/.github/workflows/docker.yml#L26-L39

**Comment by lemunozm** [comment]
@mikiquantum I've tried your solution without success in #1245. Any idea? If not, I'll try to get deep into it when I have time.

**Comment by lemunozm** [comment]
Rewards stability tests seem to be worst in debug. I propose to wait merging this change until  #1253 is done.

**Review by lemunozm — APPROVED** [review]

**Comment by lemunozm** [comment]
@mikiquantum @NunoAlexandre we need our blessings as a code owners for this

**Review by mikiquantum — APPROVED** [review]
nice!

**Comment by mustermeiszer** [comment]
On GitHub this leaves our main branch as failing, or my view is not correct.

Cc: @lemunozm @branan @mikiquantum

**Comment by lemunozm** [comment]
Yes, you're right. I think we have an issue with `rust-cache` exceeding the maximum limit. I'll try to fix it. We can revert this if needed

**Comment by branan** [comment]
I think we should just revert until we get the cache size figured out.
